h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -409,17 +402,44 @@ public void importFiles(String path, String pattern, ArrayList<String> files, Ar
       I[Value.HDFS].importFiles(path, pattern, files, keys, fails, dels);
     }
 
-    final String[] patterns = pattern != null && !pattern.isEmpty() ? 
-            ArrayUtils.append(importFilesPatterns, pattern) : importFilesPatterns; 
-    for (String p : patterns) {
-      if (p != null && !p.isEmpty())
-        filterFiles(p, path, files, keys, fails);
+    if (pattern != null && !pattern.isEmpty())
+      filterFiles((prefix, elements) -> matchPattern(prefix, elements, pattern), path, files, keys, fails);
+    filterMetadataFiles(path, files, keys, fails);
+  }
+
+  private void filterMetadataFiles(String path, ArrayList<String> files, ArrayList<String> keys, ArrayList<String> fails) {
+    filterFiles(new DeltaLakeMetadataFilter(), path, files, keys, fails);
+  }
+
+  static class DeltaLakeMetadataFilter implements Function2<String, ArrayList<String>, ArrayList<String>> {
+    @Override
+    public ArrayList<String> apply(String prefix, ArrayList<String> ids) {
+      if (isDeltaLogPath(prefix))
```

Comment:
An obscure edge case... What if a user folder is `/a/_delta_log/b` and persists data to this folder with delta. IMHO, reading from this folder would fail on files from `/a/_delta_log/b/_delta_log`.

Reply:
good point, I wanted to do this to avoid polluting the logs but I will improve it

Reply:
you are right, I think the use case of trying to read the metadata files directly doesn't make sense and is creating weird edge cases

I will fix it and ignore the files always